app/src/main/java/org/oppia/android/app/administratorcontrols/administratorcontrolsitemviewmodel/AdministratorControlsAccountActionsViewModel.kt

```diff
@@ -1,29 +1,21 @@
 package org.oppia.android.app.administratorcontrols.administratorcontrolsitemviewmodel
 
-import android.content.Intent
-import androidx.appcompat.app.AlertDialog
 import androidx.fragment.app.Fragment
-import org.oppia.android.R
-import org.oppia.android.app.shim.IntentFactoryShim
+import androidx.lifecycle.ViewModel
+import org.oppia.android.app.administratorcontrols.AdministratorControlsFragment
+import org.oppia.android.app.administratorcontrols.LogoutDialogFragment
+
+private const val TAG_LOGOUT_DIALOG = "TAG_LOGOUT_DIALOG"
 
 /** [ViewModel] for the recycler view in [AdministratorControlsFragment]. */
 class AdministratorControlsAccountActionsViewModel(
-  private val fragment: Fragment,
-  private val intentFactoryShim: IntentFactoryShim
+  private val fragment: Fragment
 ) : AdministratorControlsItemViewModel() {
 
   fun onLogOutClicked() {
-    AlertDialog.Builder(fragment.context!!, R.style.AlertDialogTheme)
-      .setMessage(R.string.log_out_dialog_message)
-      .setNegativeButton(R.string.log_out_dialog_cancel_button) { dialog, _ ->
-        dialog.dismiss()
-      }
-      .setPositiveButton(R.string.log_out_dialog_okay_button) { _, _ ->
-        // TODO(#762): Replace [ProfileChooserActivity] to [LoginActivity] once it is added.
-        val intent = intentFactoryShim.createProfileChooserActivityIntent(fragment.activity!!)
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-        fragment.activity!!.startActivity(intent)
-        fragment.activity!!.finish()
-      }.create().show()
+    LogoutDialogFragment().showNow(
+      fragment.requireActivity().supportFragmentManager,
```

Comment:
In the future, we may probably want to add other logic with LogoutDialogFragment, so suggest making a method to provide the LogOutDialogFragment instance inside a Companion Object. You can see other DialogFragment for reference.

Reply:
Done

Reply:
This can be resolved now :)